### ISSUE TITLE
Discussion about rank consistency

### ISSUE THREAD

**Issue by pprp** [pprp/PicoNAS#5]
Tricks: 

1. clear grad

https://github.com/pprp/spos-mae-cifar/blob/ae21785ab292fe417299083482e28a50ba50621a/pplib/trainer/macro_trainer.py#L67-L70

2. cancel weight decay for bn 

https://github.com/xiaomi-automl/MixPath/blob/38e601d0282cf0c82982cce0484f19f6afc684ff/NasBench101/nas_train_search.py#L164-L172

```
def separate_bn_params(model):
    bn_index = []
    bn_params = []
    for m in model.modules():
        if isinstance(m, nn.BatchNorm2d):
            bn_index += list(map(id, m.parameters()))
            bn_params += m.parameters()
    base_params = list(filter(lambda p: id(p) not in bn_index, model.parameters()))
    return base_params, bn_params
```
